### PR TITLE
Add/turret flipping

### PR DESCRIPTION
**Pull request by NolanCampbell2357**

### CODE REVIEW

File: robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java [hunk 1]
```diff
@@ -133,6 +133,19 @@ public void periodic() {
...
     }
 
+    /**
+     * Determines what direction the turret needs flipped
+     * @return Clockwise or CounterClokwise soft limit
+     */
+    public double getTurretFlipRotations() {
+        if (m_targetMotorRotations - m_config.m_turretRotationsClockwiseSoftLimit > 
+            m_targetMotorRotations - m_config.m_turretRotationsCounterClockwiseSoftLimit) {
+           return m_config.m_turretRotationsClockwiseSoftLimit;
```
Comment (coderkevin) [inline robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java:143]: Why is this returning the soft limit? Shouldn't it be returning a value plus or minus the number of motor rotations that equals 360 degrees

**Review by coderkevin — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java (hunk 1 above)
Comment (NolanCampbell2357) [reply robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java]: I fixed this.

File: robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java [hunk 2]
```diff
@@ -133,6 +133,19 @@ public void periodic() {
         SmartDashboard.putNumber("Turret Motor Output", m_turretMotor.getAppliedOutput());
     }
 
+    /**
+     * Determines what direction the turret needs flipped
+     * @return Clockwise or CounterClokwise soft limit
+     */
+    public double getTurretFlipRotations() {
+        if (m_targetMotorRotations - m_config.m_turretRotationsClockwiseSoftLimit > 
+            m_targetMotorRotations - m_config.m_turretRotationsCounterClockwiseSoftLimit) {
```
[the remark this answers is not in the document]
Comment (NolanCampbell2357) [reply robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java]: I fixed this.

File: robotbase/src/main/java/com/team2357/frc2022/commands/TurretFlipCommand.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.team2357.lib.commands.CommandLoggerBase;
+
+public class TurretFlipCommand extends CommandLoggerBase {
+    public TurretFlipCommand (TurretSubsystem turretSub) {
+        this.andThen(new TurretSetRotationCommand(turretSub, turretSub.getTurretFlipRotations()));
+    }
+
+    @Override
+    public boolean isFinished() {
+        return true;
```
[the remark this answers is not in the document]
Comment (NolanCampbell2357) [reply robotbase/src/main/java/com/team2357/frc2022/commands/TurretFlipCommand.java:13]: I am scheduling a command to flip off this command, so this one can end.

**Pull request closed without merging**